Sources/Dejavu/GRDBSession.swift

```diff
@@ -32,12 +32,19 @@ final class GRDBSession: DejavuSession, @unchecked Sendable {
         }
         
         @discardableResult
-        mutating func unregister(_ request: Request) -> Int {
+        mutating func unregister(_ request: Request, instanceCount: Int) -> Int {
             guard var instanceCount = instanceCounts[request.hashString] else {
                 return 0
             }
             instanceCount -= 1
             instanceCounts[request.hashString] = instanceCount
+
+            // decrement any higher instance counts with matching request hash
+            //
+            transactions
+                .filter { $0.value.dejavuRequest.hashString == request.hashString && $0.value.instanceCount > instanceCount }
+                .forEach { transactions[$0.key]?.instanceCount -= 1 }
```

Comment:
Without this change, any inflight requests wouldn't have their instance counts adjusted following the failed request that was unregistered